Adding coverage for usergroup entity for API

oshtaier opened the pull request:
Closes #3372

```
nosetests tests/foreman/api/test_usergroup.py
.................
----------------------------------------------------------------------
Ran 17 tests in 184.778s

OK
```

abalakh commented on tests/foreman/api/test_usergroup.py:
```diff
@@ -0,0 +1,275 @@
+            with self.subTest(new_name):
+                user_group.name = new_name
+                with self.assertRaises(HTTPError):
+                    user_group = user_group.update(['name'])
```
```python
                with self.assertRaises(HTTPError):
                    user_group.update(['name'])
```
`update()` will raise an exception, so no need in trying to set the value to some variable here.

abalakh commented on tests/foreman/api/test_usergroup.py:
```diff
@@ -0,0 +1,275 @@
+        new_user_group = entities.UserGroup().create()
+        new_user_group.name = name
+        with self.assertRaises(HTTPError):
+            new_user_group = new_user_group.update(['name'])
```
same here

abalakh commented on tests/foreman/api/test_usergroup.py:
```diff
@@ -0,0 +1,275 @@
+        """
+        new_role = entities.Role().create()
+        user_group = entities.UserGroup().create()
+        user_group.role.append(new_role)
```
I'm curious why you started using `append` instead of simple `user_group.role = [new_role]`?
Think of what will happen, for example, if usergroup will have 1 pre-created role? With ``user_group.role = [new_role]` you could just assert the first element of the list (like you did), but with `append` as you're adding the element to the end of list you should either use `assertIn` or check the latest element (`user_group.role[-1]`). Imo better to keep using our common `user_group.role = [new_role]`.

abalakh commented:
ACK, comments are not blockers

oshtaier commented:
```
nosetests tests/foreman/api/test_usergroup.py -m test_positive_update
.....
----------------------------------------------------------------------
Ran 5 tests in 26.958s

OK
```

elyezer commented on tests/foreman/api/test_usergroup.py:
```diff
@@ -0,0 +1,275 @@
+Each ``APITestCase`` subclass tests a single URL. A full list of URLs to be
+tested can be found here:
+http://theforeman.org/api/1.11/apidoc/v2/usergroups.html
+
```
Feel free to drop this blank line.

elyezer commented on tests/foreman/api/test_usergroup.py:
```diff
@@ -0,0 +1,275 @@
+        users = [entities.User().create() for _ in range(randint(3, 5))]
+        user_group = entities.UserGroup(user=users).create()
+        self.assertEqual(
+            set(user.login for user in users),
```
I would not use set here, but a sorted list.

Set can mask a failure when an user is added two times for example.

oshtaier commented:
Addressed comments
```
nosetests tests/foreman/api/test_usergroup.py -m test_positive_create
.......
----------------------------------------------------------------------
Ran 7 tests in 122.231s

OK
```

elyezer commented:
ACK
